{{cookiecutter.repo_name}}/devtools/scripts/create_conda_env.py

```diff
@@ -0,0 +1,96 @@
+import argparse
+import os
+import re
+import shutil
+import subprocess as sp
+from tempfile import NamedTemporaryFile
+
+# Args
+parser = argparse.ArgumentParser(description='Creates a conda environment from file for a given Python version.')
+parser.add_argument('-n', '--name', type=str,
+                    help='The name of the created Python environment')
+parser.add_argument('-p', '--python', type=str,
+                    help='The version of the created Python environment')
+parser.add_argument('conda_file',
+                    help='The file for the created Python environment')
+
+args = parser.parse_args()
+
+# Open the base file
+with open(args.conda_file, "r") as handle:
+    script = handle.read()
+
+# Make changes based on the options (not relying on YAML)
+# Determine whitespace level of first non-commented line
+try:
+    whitespace = re.search('^[\r\n]*?([ \t\f]*)[^\#\s]', script, re.MULTILINE).group(1)
+except AttributeError:
+    raise RuntimeError("The input Conda Environment File is either all comments or not YAML formatted. "
+                       "Please check your file")
+# Python
+python_replacement_string = "- python {}*".format(args.python)
+dep_entry = re.search('^{}dependencies:'.format(whitespace), script, re.MULTILINE)
+if not dep_entry:
+    # Case of no dependencies entry
+    script += '\n{ws}dependencies:\n{ws} {py}'.format(ws=whitespace, py=python_replacement_string)
+else:
+    # Calculate the end of the dependencies
+    try:
+        dep_span = re.search('^[\r\n]*{}[^\#\s-]'.format(whitespace), script[dep_entry.end():], re.MULTILINE).start()
+    except AttributeError:
+        dep_span = None
+    # Calculate whitespace of dependencies
+    dep_whitespace_re = re.search('^[\r\n]*?{}([ \t\f]*)-'.format(whitespace),
+                                  script[dep_entry.end():dep_span],
+                                  re.MULTILINE)
+    if dep_whitespace_re is None:
+        # Dependencies entry, but no values
+        dep_block = '\n{ws} {py}\n'.format(ws=whitespace, py=python_replacement_string)
+    else:
+        dep_block = script[dep_entry.end():dep_span]
+        dep_whitespace = dep_whitespace_re.group(1)
+        if not re.search('^[\r\n]*?{}{}-[ \t]*python'.format(whitespace, dep_whitespace), dep_block,
+                         re.MULTILINE):
+            # Dependencies but no uncommented python
+            dep_block += '\n{ws}{dws}{py}\n'.format(ws=whitespace, dws=dep_whitespace, py=python_replacement_string)
+        else:
+            # Finally: Dependencies with fields and one of them is python
+            dep_block = re.sub(r'-[ \t]*python([ ><=*]+[0-9.*]*)?$', python_replacement_string, dep_block,
+                               flags=re.MULTILINE)
+    final_script = script[:dep_entry.end()] + dep_block
+    if dep_span is not None:
+        final_script += script[dep_span:]
+    script = final_script
+
```

Comment:
I'm normally regex-happy, but I'll register a slight vote for contortions to get yaml in env rather than the voluminous regex alternative.

Reply:
I started leaning that way about half way through, but wanted to get a complete answer so everyone could see what it looks like and offer input.

Reply:
I vote a bit for YAML as well (we could also fall back to regex?). But no real preference as long as it works.

Either way let’s settle on a solution today so that we can run some trials of this Wednesday.

Reply:
Whoa, this looks quite a bit harder to maintain.
What's the problem with YAML?